plugins/qcheck-stm/src/reserr.mli

```diff
@@ -49,6 +49,9 @@ val promote : 'a reserr list -> 'a list reserr
     no [errors] of level [Error] in [rs] and store the [errors] of level
     [Warning] in the warnings list *)
 
+val promote_opt : 'a reserr -> 'a option reserr
+(** [promote_opt r] is [promote \[ r \] ] *)
```

Comment:
That documentation is bending types but maybe that’s close enough?
(Otherwise, this could be written as
```[promote_opt] is [promote] for a unique value```

Reply:
You're right, let'b be type correct!